imap_processing/swapi/l1/swapi_l1.py

```diff
@@ -1 +1,565 @@
+    # to this:
+    # [
+    #   0  0  0  0  0  0
+    #   1  1  1  1  1  1
+    #   2  2  2  2  2  2
+    #   3  3  3  3  3  3
+    #   4  4  4  4  4  4
+    #   5  5  5  5  5  5
+    #   6  6  6  6  6  6
+    #   7  7  7  7  7  7
+    #   8  8  8  8  8  8
+    #   9  9  9  9  9  9
+    #   10 10 10 10 10 10
+    #   11 11 11 11 11 11
+    # ]
+
+    pcem_count = np.ravel(pcem_count, order="F")
+    pcem_compressed = np.ravel(pcem_rng_val, order="F")
+    scem_count = np.ravel(scem_count, order="F")
+    scem_compressed = np.ravel(scem_rng_val, order="F")
+    coin_count = np.ravel(coin_count, order="F")
+    coin_compressed = np.ravel(coin_rng_val, order="F")
+    return (
+        pcem_count,
+        scem_count,
+        coin_count,
+        pcem_compressed,
+        scem_compressed,
+        coin_compressed,
+    )
+
+
+def process_swapi_science(sci_dataset):
+    """Process SWAPI science data.
+
+    Parameters
+    ----------
+    dataset : xarray.Dataset
+        L0 data
+    """
+    full_sweep_indices = get_indices_of_full_sweep(sci_dataset["SEQ_NUMBER"].data)
+    full_sweep_sci = filter_full_cycle_data(full_sweep_indices, sci_dataset)
+    swp_pcem_counts = []
+    swp_scem_counts = []
+    swp_coin_counts = []
+    swp_pcem_comp = []
+    swp_scem_comp = []
+    swp_coin_comp = []
+    collections.defaultdict(list)
+
+    total_packets = len(full_sweep_sci["SEQ_NUMBER"].data)
+
+    # Step through each twelve packets that makes full sweep
+    for sweep_index in range(0, total_packets, 12):
+        m = sweep_index
+        n = sweep_index + 12
+        # If PLAN_ID and SWEEP_TABLE is not same, the discard the
+        # sweep data. PLAN_ID and SWEEP_TABLE should match
+        if np.all(full_sweep_sci["PLAN_ID_SCIENCE"].data[m:n]) and np.all(
+            full_sweep_sci["SWEEP_TABLE"].data[m:n]
+        ):
+            # TODO: add log here
+            continue
+        # TODO: add other checks for bad data
+        # Process good data.
+        # full data will be of
+        (
+            pcem_counts,
+            scem_counts,
+            coin_counts,
+            pcem_comp,
+            scem_comp,
+            coin_comp,
+        ) = create_full_sweep_data(full_sweep_sci, sweep_index)
+
+        swp_pcem_counts.append(pcem_counts)
+        swp_scem_counts.append(scem_counts)
+        swp_coin_counts.append(coin_counts)
+        swp_pcem_comp.append(pcem_comp)
+        swp_scem_comp.append(scem_comp)
+        swp_coin_comp.append(coin_comp)
+
+    # Get Epoch time of full sweep data and then reshape it to
+    # (n, 12) where n = total number of full sweep data and 12 = 12
+    # sequence data's metadata. For Epoch's data, we take the first element
+    # of each 12 sequence data's metadata.
+    epoch_time = xr.DataArray(
+        sci_dataset["Epoch"].data[full_sweep_indices].reshape(-1, 12)[:, 0],
+        name="Epoch",
+        dims=["Epoch"],
+    )
+    counts = xr.DataArray(np.arange(72), name="Counts", dims=["Counts"])
```

Comment:
Is this dimension counts, or should it be "step" or "sweep" number? Because I think the counts are the variable, but the dimension is the independent variable that the data is a function of.